src/main/java/org/gridsuite/study/server/elasticsearch/EquipmentInfosServiceImpl.java

```diff
@@ -56,10 +56,37 @@ public void deleteAll(@NonNull UUID networkUuid) {
     @Override
     public List<EquipmentInfos> search(@NonNull final String query) {
         NativeSearchQuery nativeSearchQuery = new NativeSearchQueryBuilder()
-                .withQuery(QueryBuilders.queryStringQuery(query))
-                .withPageable(PageRequest.of(0, PAGE_MAX_SIZE))
-                .build();
-        SearchHits<EquipmentInfos> searchHits = elasticsearchOperations.search(nativeSearchQuery, EquipmentInfos.class);
-        return searchHits.stream().map(SearchHit::getContent).collect(Collectors.toList());
+            .withQuery(QueryBuilders.queryStringQuery(query))
+            .withPageable(PageRequest.of(0, PAGE_MAX_SIZE))
+            .build();
+
+        Map<String, List<EquipmentInfos>> infos = elasticsearchOperations.search(nativeSearchQuery, EquipmentInfos.class)
+            .stream()
+            .map(SearchHit::getContent)
+            .collect(Collectors.groupingBy(EquipmentInfos::getType));
+
+        // Sorted by equipment type
+        return Stream.of(
+                infos.getOrDefault(EquipmentType.SUBSTATION.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.VOLTAGE_LEVEL.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.LINE.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.TWO_WINDINGS_TRANSFORMER.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.THREE_WINDINGS_TRANSFORMER.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.HVDC.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.GENERATOR.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.BATTERY.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.LOAD.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.SHUNT_COMPENSATOR.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.DANGLING_LINE.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.STATIC_VAR_COMPENSATOR.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.HVDC_CONVERTER_STATION.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.BUSBAR_SECTION.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.BREAKER.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.DISCONNECTOR.name(), Collections.emptyList()),
+                infos.getOrDefault(EquipmentType.LOAD_BREAK_SWITCH.name(), Collections.emptyList())
+            )
+            .flatMap(Collection::stream)
+            .sorted(Comparator.comparing(EquipmentInfos::getId))
+            .collect(Collectors.toList());
```

Comment:
This part of code seems to return a list sorted by equipment id, but not by equipment type as stated in the comment.